### ISSUE TITLE
Publish cube to Lindas DB

### ISSUE THREAD

**Issue by martinmaillard** [zazuko/cube-creator#229]

**Comment by lucafurrer** [comment]
Questions
- Where will the Endpoint be defined? UI or Environment variable?
- Will the last Cube be published, or do we have any history to choose from?

**Comment by martinmaillard** [comment]
> - Where will the Endpoint be defined? UI or Environment variable?

Environment variable

> - Will the last Cube be published, or do we have any history to choose from?

We don't have a history so depending on how the publication process works it's either:
1. Publish the current version of the cube (if it's a separate job)
2. Publish the job that is currently being transformed (if it's part of the same pipeline than the transformation)

**Comment by lucafurrer** [comment]
What goes into Lindas
- cubeProject cc:cubeGraph 
- cubeProject cc:dataset dataset
  - dataset cc:dimensionMetadata
  - dataset schema:hasPart
  - dataset * (cube meta data)

How will this be added together to the target place? 
What happens with the dimension metadata?
How do schema:hasPart and cubeGraph fit together?

Is it the CLI which handles the copy of the graph or is it core?

**Comment by martinmaillard** [comment]
I'll try to answer with what I think I know:

- When the transform pipeline runs, it already brings together all the parts you mention to form the final cube. Only the final cube needs to be copied to lindas.
- Publication will be a type of Job (like Transform). So I guess it will be created on the `JobCollection` of the project.

**Comment by lucafurrer** [comment]
And a technical question, is there a library which helps to copy a graph from one store to an other?

**Comment by tpluscode** [comment]
>  is there a library which helps to copy a graph from one store to an other?

As hinted in the previous comments, we would use a Barnard59 pipeline. This is the same toolset which is used by the transformation jobs.

TL;DR; It would go like this:

1. create a stream of triples from our store
2. write them to a graph using [SPARQL Graph Store Protocol](https://www.w3.org/TR/sparql11-http-rdf-update/)
